### DISCUSSION TITLE
Make goals easier to set up for forms and buttons

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#435] [Anything Else]
"The outbound link clicking thing looks way easier. Obviously it's easier to implement because you can just look at the actual outbound link address, not really something you can do for the buttons. But for the buttons and submission forms, the ideal user experience (probably difficult to implement but I'm just saying from a user perspective) for me would be something like this :

1. I click "create new trigger on this website"
2. Name this trigger ("form submission")
2. It opens the website in the background, scrollable, says "click on the element you want to track. must be button or form element"
3. I click on the thing (selectable elements are highlighted when hovered)
4. Finished. "Open site in new page and click on this button to see first click tracked!""

(feedback via email)

**Comment by ukutaht** [comment]
I agree that we should make form & internal link/button tracking easier.

The UI explained here sounds great for the user but it's very difficult to build.

A much simpler approach would be to offer an extension to the default tracking script, say: `<script src="https://plausible.io/js/plausible.forms.js">` and then the user would have to tag each form the want to track with `<form data-plausible-event="Signup Form: Submit">`

You need to be able to control the HTML of your website so more limited CMS tools wouldn't work with this approach.
